### PR TITLE
#1017 Implemented and tested BitbucketCommits/Commit.

### PR DESCRIPTION
**Pull request by criske**
- Left puzzle for BitbucketCommit#comments.

PR for #1017

### CODE REVIEW

**Comment by zoeself** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by zoeself** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2021-03-03T11:07:17.767511``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: self-core-impl/src/main/java/com/selfxdsd/core/BitbucketCommit.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * {@inheritDoc}
+     * <br/>
+     * Since Bitbucket 2.0 doesn't reveal the `username` anymore in their json
+     * responses, we will extract the `account_id` as username from `author`
+     * section.
+     */
```
Comment (criske) [inline self-core-impl/src/main/java/com/selfxdsd/core/BitbucketCommit.java:101]: @amihaiemil also when add contributors/contracts we must register them by their `account_id`. Basically PO must invite by account_id on `self-web` and by email on bitbucket.org. I know, it's weird :D.

Comment (amihaiemil) [reply self-core-impl/src/main/java/com/selfxdsd/core/BitbucketCommit.java:101]: @criske well, I guess we will fetch the accountId based on the username that the PO gives as input in the Add New Contract formular :D 

We'll see, it's ok for now.

**Comment by amihaiemil** [comment]
@criske looks good, thanks!

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by zoeself** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**